### PR TITLE
Migrate CI to Github actions for master branch

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR moves all the CI to GitHub actions for the master branch.

### CODE REVIEW

File: .github/workflows/ci-master.yml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+              run: php -v
+
+            - name: Composer install
+              run: composer install --optimize-autoloader --prefer-dist
+
+            - name: Install WP Tests
+              run: bash tests/bin/install-wp-tests.sh wordpress_test root root localhost latest
+
+            - name: Run PHP Unit Tests
+              run: phpunit --configuration tests/phpunit.xml
```
Comment (tn3rb) [inline .github/workflows/ci-master.yml:115]: love how this is all laid out... much easier to follow than Travis, although there does appear to be some extra complexity

File: .github/workflows/ci-master.yml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+              # Allow failure for nightly builds - PHP 8.0/8.1
+              continue-on-error: ${{ matrix.php-versions == '8.0' || matrix.php-versions == '8.1' }}
```
Comment (tn3rb) [inline .github/workflows/ci-master.yml:117]: 👍

File: .github/workflows/ci-master.yml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+              # run multisite test only with 1 PHP version
+            - if: ${{ matrix.php-versions == '7.4' }}
+              name: Run PHP Unit Tests - WP Multisite
```
Comment (tn3rb) [inline .github/workflows/ci-master.yml:121]: 👍

File: .github/workflows/ci-master.yml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+              env:
+                  WP_MULTISITE: 1
+              run: phpunit --configuration tests/phpunit.xml
+
+    js-lint-and-build:
+        name: JS Lint/Build
+        runs-on: ubuntu-latest
+        needs: changed
+        # Run only if files in /assets/ have changed
+        if: ${{ needs.changed.outputs.assets == 'true' }}
```
Comment (tn3rb) [inline .github/workflows/ci-master.yml]: THIS IS AWESOME !!!

File: .github/workflows/ci-master.yml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            - name: PHP Lint
+              run: composer config-eventespressocs && composer run-script lint:skip-warnings
+              # Allow failure for PHP > 7.3
+              continue-on-error: ${{ matrix.php-versions != '7.3' }}
+
+    php-unit-tests:
+        runs-on: ${{ matrix.operating-system }}
+        needs: changed
+        # Run only if PHP files have changed
+        if: ${{ needs.changed.outputs.php == 'true' }}
```
Comment (tn3rb) [inline .github/workflows/ci-master.yml]: SUPER COOL !!!

**Review by tn3rb — APPROVED** [review]
# SUPER AWESOME

only thing that isn't 1000% amazing is that we don't need CodeSniffer linting to run on multiple versions of PHP, one is enough.

**Review by nicolad — APPROVED** [review]
LGTM

**Review by tn3rb — APPROVED** [review]

**Pull request merged**